addons/web/static/tests/views/pivot_view.test.js

```diff
@@ -0,0 +1,4551 @@
+        type: "pivot",
+        resModel: "partner",
+        arch: `
+			<pivot>
+				<field name="date" type="row" interval="day"/>
+				<field name="bar" type="col"/>
+				<field name="bar" string="SLA status Failed" type="measure"/>
+			</pivot>`,
+    });
+
+    expect('tbody tr:contains("2019-12-14")').toHaveCount(1);
+    expect('tbody tr:contains("2019-12-14") [type="checkbox"]').toBeChecked();
+    expect('tbody tr:contains("2019-05-14")').toHaveCount(1);
+    expect('tbody tr:contains("2019-05-14") [type="checkbox"]').not.toBeChecked();
+});
+
+test.tags("desktop")("empty pivot view with action helper", async () => {
+    Partner._views["pivot,false"] = `<pivot>
+		<field name="product_id" type="measure"/>
+		<field name="date" interval="month" type="col"/>
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        context: { search_default_small_than_0: true },
+        noContentHelp: markup(`<p class="abc">click to add a foo</p>`),
+        config: {
+            views: [[false, "search"]],
+        },
+    });
+
+    expect(".o_view_nocontent .abc").toHaveCount(1);
+    expect("table").toHaveCount(0);
+    await removeFacet();
+    expect(".o_view_nocontent .abc").toHaveCount(0);
+    expect("table").toHaveCount(1);
+});
+
+test.tags("desktop")("empty pivot view with sample data", async () => {
+    Partner._views["pivot,false"] = `<pivot sample="1">
+		<field name="product_id" type="measure"/>
+		<field name="date" interval="month" type="col"/>
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        context: { search_default_small_than_0: true },
+        noContentHelp: markup('<p class="abc">click to add a foo</p>'),
+        config: {
+            views: [[false, "search"]],
+        },
+    });
+
+    expect(".o_pivot_view .o_content").toHaveClass("o_view_sample_data");
+    expect(".o_view_nocontent .abc").toHaveCount(1);
+    await removeFacet();
+    expect(".o_pivot_view .o_content").not.toHaveClass("o_view_sample_data");
+    expect(".o_view_nocontent .abc").toHaveCount(0);
+    expect("table").toHaveCount(1);
+});
+
+test("non empty pivot view with sample data", async () => {
+    Partner._views["pivot,false"] = `<pivot sample="1">
+		<field name="product_id" type="measure"/>
+		<field name="date" interval="month" type="col"/>
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        noContentHelp: markup('<p class="abc">click to add a foo</p>'),
```

Comment:
@odoo/rd-security The string in the function is a constant. What's more, the function is used in a unit test. The risk is zero.